data/org.freedesktop.portal.Secret.xml

```diff
@@ -0,0 +1,61 @@
+<?xml version="1.0"?>
+<!--
+ Copyright (C) 2019 Red Hat, Inc.
+
+ This library is free software; you can redistribute it and/or
+ modify it under the terms of the GNU Lesser General Public
+ License as published by the Free Software Foundation; either
+ version 2 of the License, or (at your option) any later version.
+
+ This library is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ Lesser General Public License for more details.
+
+ You should have received a copy of the GNU Lesser General Public
+ License along with this library. If not, see <http://www.gnu.org/licenses/>.
+
+ Author: Daiki Ueno <dueno@redhat.com>
+-->
+
+<node name="/" xmlns:doc="http://www.freedesktop.org/dbus/1.0/doc.dtd">
+  <!--
+      org.freedesktop.portal.Secret:
+      @short_description: Portal for retrieving application secret
+
+      The Secret portal allows sandboxed applications to retrieve a
+      per-application secret.  The secret can then be used for
+      encrypting confidential data inside the sandbox.
+
+      This documentation describes version 1 of this interface.
+  -->
+  <interface name="org.freedesktop.portal.Secret">
+
+    <!--
+        RetrieveSecret:
+        @fd: Writable file descriptor for transporting the secret
+        @options: Vardict with optional further information
+        @handle: Object path for the #org.freedesktop.portal.Request object representing this call
+
+        Retrieves a master secret for a sandboxed application.
+
```

Comment:
We should document some guarantees about the 'master secret' here:

- It is unique per application

- It does not change as long as the application is installed (once it has been created)

- It is stored in the users keyring, under the application ID

- What is the format of this secret ? It would be useful to know what data is actually returned here. Or is this always used via a library api that hides the details ?

Reply:
If libsecret has some docs about this, then just saying something like "in the format understood by libsecret" might be good enough

Reply:
Also the options it accepts should be documented.